src/helper_functions.jl

```diff
@@ -525,3 +527,118 @@ end
 
 end # @eval
 end # if
+
+"""
+    subtypes(interface, concrete::Bool)
+
+Return the concrete subtypes of a given interface.
+
+### Input
+
+- `interface` -- an abstract type, usually a set interface
+- `concrete`  -- if `true`, seek further the inner abstract subtypes of the given
+                 interface, otherwise return only the direct subtypes of `interface`
+
+### Output
+
+A list with the subtypes of the abstract type `interface`, sorted alphabetically.
+
+### Examples
+
+Consider the `AbstractPolytope` interface. If we include the abstract subtypes
+of this interface,
+
+```jldoctest subtypes
+julia> using LazySets: subtypes
+
+julia> subtypes(AbstractPolytope, false)
+4-element Array{Any,1}:
+ AbstractCentrallySymmetricPolytope
+ AbstractPolygon
+ HPolytope
+ VPolytope
+```
+
+We can use this function to obtain the concrete subtypes of
+`AbstractCentrallySymmetricPolytope` and `AbstractPolygon` (further until all
+concrete types are obtained), using the `concrete` flag:
+
+```jldoctest subtypes
+julia> subtypes(AbstractPolytope, true)
+14-element Array{Type,1}:
+ Ball1
+ BallInf
+ HPolygon
+ HPolygonOpt
+ HPolytope
+ Hyperrectangle
+ Interval
+ LineSegment
+ Singleton
+ SymmetricIntervalHull
+ VPolygon
+ VPolytope
+ ZeroSet
+ Zonotope
+```
+
+This function can be applied to other abstract types as well:
+
+```jldoctest subtypes
+julia> subtypes(Real, false)
```

Comment:
I would drop this part because it depends too much on external packages. After all, we do not export this function.